File: MVVMDemo/koinloader-demo/src/main/java/com/hld/mvvm/koinloader_demo/TestKoinLoader2.java
package com.hld.mvvm.koinloader_demo;

import com.hld.koin.loader.KoinModule;
import com.hld.koin.loader.KoinModuleProvider;

import java.util.LinkedList;
import java.util.List;

@KoinModule
public class TestKoinLoader2 implements KoinModuleProvider {
    @Override
    public List<org.koin.core.module.Module> getModules() {
        return new LinkedList<>();
    }
}

File: MVVMDemo/koinloader-demo/src/main/java/com/hld/mvvm/koinloader_demo/TestKoinLoader2.kt
package com.hld.mvvm.koinloader_demo

import com.hld.koin.loader.KoinModule
import com.hld.koin.loader.KoinModuleProvider
import org.koin.core.module.Module
import org.koin.dsl.module

@KoinModule
class TestKoinLoader2 : KoinModuleProvider {
    override fun getModules(): List<Module> {
        return listOf(
            module {
                factory { TestDemoBean2() }
            }
        )
    }
}